.github/workflows/update-snap-revision.yaml

```diff
@@ -0,0 +1,67 @@
+name: Update Snap Revisions
+
+on:
+  schedule:
+    - cron: "0 */5 * * *" # every 5 hours
+
+
+jobs:
+  stable-branches:
+    runs-on: ubuntu-latest
+    outputs:
+      branches: ${{ steps.release-branches.outputs.data }}
+    steps:
+    - uses: octokit/request-action@v2.x
+      id: list-branches
+      with:
+        route: GET /repos/${{ github.repository }}/branches
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    - id: release-branches
+      run: |-
+        DATA='${{ steps.list-branches.outputs.data }}'
+        NAMES=$(jq -r '.[] | .name' <<< $DATA)
+        RELEASES=()
+        for BRANCH in ${NAMES}; do
+            if [[ $BRANCH == release-* ]]; then
+                RELEASES+=($BRANCH)
+            fi
+        done
+        echo data=$(printf '%s\n' "${RELEASES[@]}" | jq -R . | jq -s .) >> ${GITHUB_OUTPUT}
+    
+  update-branches:
+    runs-on: ubuntu-latest
+    needs: [stable-branches]
+    strategy:
+      matrix:
+        branch: ${{ fromJSON(needs.stable-branches.outputs.branches) }}
+        arch: ["amd64", "arm64"]
+    steps:
+    - name: Checkout ${{ matrix.branch }}
+      uses: actions/checkout@v4
+      with:
+        repository: ${{ github.repository }}
+        ref: ${{ matrix.branch }}
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.12'
+
+    - name: Update Revision
+      id: update-revision
+      run: |
+        python3 .github/workflows/update-snap-revision.py ${{ matrix.arch }} ${{ env.TRACK }} stable
```

Comment:
Just one question. Where is this TRACK envvar set?

Reply:
wow, nice call.  Updated and spotted another bug.  Thanks @mateoflorido

Reply:
track env is now from [here](https://github.com/canonical/k8s-operator/pull/83/files#diff-a2827cd6b7ce4ca1549d38bb2c36326ea0662e9e5f874a469569480ab3c9d119R40-R43)